From 73432e872068f7ea3ef93eaee294d6419af3add2 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Mon, 15 Sep 2025 10:42:20 +0200
Subject: [PATCH 150/553] QUnicodeTables: remove "wasm 64 packing trick"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

It is expected that this is no longer needed.

The upstream issue¹ is closed since 2016-09-05 (nine years ago at this
point; even though the work-around was added to Qt still two years
after upstream was fixed).

Amends e59ba35f1b1954062266164f7b802076dc152c7b (for the generator)
and a2efb8d94bb8e860d8616f56c4f40e7c41eb3810 (for the generated file;
squashed into 29c0377f07f4942f9957ea87d59c252148dc9e5b for the linear
history)

¹ https://github.com/emscripten-core/emscripten/issues/4540

Task-number: QTBUG-66621
Task-number: QTBUG-69421
Task-number: QTBUG-74511
Pick-to: 6.9 6.8 6.5
Change-Id: I9b08792aa4f5f18240d9425c44fa7a88e63e27da
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
Reviewed-by: Lorn Potter <lorn.potter@qt.io>
(cherry picked from commit 97fcd9d707b6604dae5f4edc674ab45913ed19e0)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/text/qunicodetables_p.h | 6 ------
 util/unicode/main.cpp               | 6 ------
 2 files changed, 12 deletions(-)

diff --git a/src/corelib/text/qunicodetables_p.h b/src/corelib/text/qunicodetables_p.h
index e5ba21089e6..d7b71f57d05 100644
--- a/src/corelib/text/qunicodetables_p.h
+++ b/src/corelib/text/qunicodetables_p.h
@@ -56,13 +56,7 @@ struct Properties {
     ushort unicodeVersion      : 5; /* 5 used */
     ushort eastAsianWidth      : 3; /* 3 used */
     ushort nfQuickCheck        : 8;
-#ifdef Q_OS_WASM
-    unsigned char              : 0; //wasm 64 packing trick
-#endif
     std::array<CaseConversion, NumCases> cases;
-#ifdef Q_OS_WASM
-    unsigned char              : 0; //wasm 64 packing trick
-#endif
     ushort graphemeBreakClass  : 5; /* 5 used */
     ushort wordBreakClass      : 5; /* 5 used */
     ushort lineBreakClass      : 6; /* 6 used */
diff --git a/util/unicode/main.cpp b/util/unicode/main.cpp
index 03987ec46ee..1868d6d3841 100644
--- a/util/unicode/main.cpp
+++ b/util/unicode/main.cpp
@@ -1001,13 +1001,7 @@ static const char *property_string =
     "    ushort unicodeVersion      : 5; /* 5 used */\n"
     "    ushort eastAsianWidth      : 3; /* 3 used */\n"
     "    ushort nfQuickCheck        : 8;\n" // could be narrowed
-    "#ifdef Q_OS_WASM\n"
-    "    unsigned char              : 0; //wasm 64 packing trick\n"
-    "#endif\n"
     "    std::array<CaseConversion, NumCases> cases;\n"
-    "#ifdef Q_OS_WASM\n"
-    "    unsigned char              : 0; //wasm 64 packing trick\n"
-    "#endif\n"
     "    ushort graphemeBreakClass  : 5; /* 5 used */\n"
     "    ushort wordBreakClass      : 5; /* 5 used */\n"
     "    ushort lineBreakClass      : 6; /* 6 used */\n"
-- 
2.51.2

